src/WebJobs.Extensions.CosmosDB/Trigger/CosmosDBTriggerAttributeBindingProvider.cs

```diff
@@ -137,7 +143,7 @@ public async Task<ITriggerBinding> TryCreateAsync(TriggerBindingProviderContext
                 if (attribute.CreateLeaseCollectionIfNotExists)
                 {
                     // Not disposing this because it might be reused on other Trigger since Triggers could share lease collection
-                    ICosmosDBService service = _configProvider.GetService(leasesConnectionString);
+                    ICosmosDBService service = _configProvider.GetService(leasesConnectionString, attribute.PreferredLocations);
```

Comment:
This should probably used the resolved `PreferredLocations`, which will be used for a lookup, right? Would allow sharing of the service if it's configured the same for an output/input binding.

Reply:
Good catch!